File: dotnet_entity_framework/Models/Teacher.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet_entity_framework.Models;

[Table("teacher_info")]
public class Teacher
{
    [Key, ForeignKey("User")]
    public int UserId { get; set; }
    public string ScienDegree { get; set; }
    public string BusinessEmail { get; set; }
    public string? ContactNumber { get; set; }
    public string? Room { get; set; }
    public string? ConsultationHours { get; set; }

    public User User { get; set; } = null!;
}

File: js/sequelize/Entities/Teacher.js
const {database} = require("../database");
const {DataTypes} = require('sequelize');

module.exports.Teacher = database.define('Teacher', {
    user_id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
    },
    scien_degree: {
        type: DataTypes.STRING(50),
        allowNull: false
    },
    business_email: {
        type: DataTypes.STRING,
        allowNull: false
    },
    contact_number: {
        type: DataTypes.STRING(20),
        allowNull: true
    },
    room: {
        type: DataTypes.STRING(20),
        allowNull: true
    },
    consultation_hours: {
        type: DataTypes.STRING,
        allowNull: true
    }
}, {
    tableName: 'teacher_info',
    timestamps: false
});